packages/dds/tree/src/simple-tree/schemaTypes.ts

```diff
@@ -485,16 +523,92 @@ export type TreeNodeFromImplicitAllowedTypes<
+ *
+ * Note that this is distinct from the case where the schema is `[A, B]`.
+ * In this case is it known that the field allows both A and B (the field can be set to an A or a B value).
+ * When `A | B` is used, the field might allow
+ * A but not B (so assigning a B value would be out of schema),
+ * B but not A (so assigning an A value would be out of schema)
+ * or both A and B.
+ *
+ * This gets more extreme when given completely unspecified schema.
+ * For example if a field is just provided {@link ImplicitFieldSchema}, nothing is known about the content of the field.
+ * This means reading the field (via {@link TreeFieldFromImplicitField}) can give any valid tree field content,
+ * but there are no safe values which could be written to the field (since its unknown what values would be out of schema) so {@link InsertableTreeFieldFromImplicitField} gives `never`.
+ *
+ * To implement this variance correctly, computing of types for input and output have to use separate utilities
+ * which take very different approaches when encountering non-exact schema like unions or `ImplicitFieldSchema`.
+ * The utilities which behave contravariantly as required to handle input correctly link this documentation to indicate that this is how they behave.
+ *
+ * In addition to behaving contravariantly these input type computation utilities often have further limitations.
+ * This is due to TypeScript making it difficult to implement this contravariance exactly.
+ * When faced with these implementation limitations these contravariant type computation utilities error on the side of producing overly strict requirements.
+ * For example in the above case of `A | B`, the utilities might compute an allowed insertable type as `never` even if there happens to be a common value accepted by both `A` and `B`.
+ * Future versions of the API can relax these requirements as the type computations are made more accurate.
+ *
+ * For a more concrete example: if {@link InsertableTreeFieldFromImplicitField} produced `never` for a schema `A | OptionalField<A>`,
+ * a future version could instead return a more flexible but still safe type, like `A`.
+ *
+ * More generally: try and avoid providing non-exact schema, especially for the fields of other schema.
+ * While these APIs attempt to handle such cases correctly, there are limitations and known bugs in this handling.
+ * Code using non-exact schema is much more likely to have its compilation break due to updates of this package or even TypeScript,
+ * and thus compilation breaks to edge cases of non-exact schema handling, especially with recursive schema, are not considered breaking changes.
+ * This may change as the API become more stable.
+ *
+ * @privateRemarks
+ * There likely is a better way to share this documentation, but none was found at the time of writing.
```

Comment:
I wonder if there are JSDoc/TSDoc pre-processor tools out there that we could utilize to add custom behaviors like `{@include <some-target-docs-definition>}` and actually modify the doc contents during TSC's build. If the `d.ts` files then have properly TSDoc-compliant contents, we wouldn't have to worry about incompatibility with API-Extractor.

Alternatively, if one doesn't exist already, writing one probably wouldn't be all too difficult and might be a good FHL project.

Reply:
Something like `include` would also be great for generating example code. I would love to check in examples as "test" code that we build and have them injected into our docs automagically.

Reply:
We could also use such a tool to generate blocks of contents for our existing custom tags like `@system` and `@legacy` so that end-user intellisense would include relevant notices.

Reply:
I'd love to have rust's doc tests, and ability to have file include macros (even in docs)...

I think making a "tag -> notice" injector as a build or publish step to match the API docs site would be the cleanest and most consistent. Would be great for system, legacy, deprecated, but could be used for other adhock cases like this.

Alternatively finding a way to link markdown files from /docs could also meet our needs here.

Reply:
This can theoretically be done, but it's a bit fragile. E.g.
```typescript
/**
 * @see {@link https://github.com/microsoft/FluidFramework/blob/main/docs/README.md}
 */
```